In [ ]:
# We import the necessary libraries to download and manipulate the data, and to visualize it.
import pandas as pd
import yfinance as yf

In [ ]:
# # Moveing the variables to the top of the notebook and made it consistent with the rest of the notebooks
# ASSET = "BTC-USD"
# START_DATE = "2018-01-01"
# END_DATE = pd.Timestamp.now()
# INTERVAL = "1d"

In [ ]:
# Removing the variables to the top of the notebook and use the config
from config import ASSETS, START_DATE, END_DATE, INTERVAL

In [ ]:
# Download Bitcoin data from yfinance from 2018-01-01 to actual date time, and interval of 1 day
btc_data = yf.download(
    start = START_DATE, 
    end = END_DATE, 
    tickers = ASSETS,
    interval = INTERVAL)

[*********************100%***********************]  1 of 1 completed


In [10]:
# We check the data we downloaded
btc_data.info()
btc_data.describe()
btc_data.isnull().sum()
btc_data.head()
btc_data.tail()

<class 'pandas.DataFrame'>
DatetimeIndex: 3074 entries, 2018-01-01 to 2026-06-01
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   (Close, BTC-USD)   3073 non-null   float64
 1   (High, BTC-USD)    3073 non-null   float64
 2   (Low, BTC-USD)     3073 non-null   float64
 3   (Open, BTC-USD)    3073 non-null   float64
 4   (Volume, BTC-USD)  3074 non-null   int64  
dtypes: float64(4), int64(1)
memory usage: 144.1 KB


Price,Close,High,Low,Open,Volume
Ticker,BTC-USD,BTC-USD,BTC-USD,BTC-USD,BTC-USD
Date,,,,,
2026-05-28,73536.554688,74460.125000,72493.414062,74339.570312,40148145327
2026-05-29,73372.523438,74218.562500,72435.625000,73537.031250,34457929250
2026-05-30,73754.835938,74020.757812,73125.234375,73370.851562,19563589191
2026-05-31,73579.687500,74153.500000,73315.015625,73753.750000,17445435650
2026-06-01,NaN,NaN,NaN,NaN,43300679680


In [11]:
# Make it csv and save it in or raw data folder
btc_data.to_csv("../data/raw/btc_data_1d.csv")

# Dataset Summary

### Source
Yahoo Finance (BTC-USD)

### Time Range
2018-01-01 to 2026-06-01

### Frequency
Daily

### Number of Observations
3074

### Features
- Open
- High
- Low
- Close
- Volume

### Missing Values
No missing values were detected.

### Data Structure
The dataset uses a MultiIndex column structure generated by yfinance. The second level corresponds to the asset ticker (BTC-USD).

### Initial Assessment
The dataset contains sufficient historical information to capture multiple market conditions, including bullish, bearish, and high-volatility periods. No immediate data quality issues were detected.

## Future Scalability Considerations

The current analysis focuses on BTC-USD. However, the project architecture should support future expansion to multiple assets.

Potential configurable parameters:

- Assets
- Start date
- End date
- Sampling frequency

Future versions should minimize hardcoded values and allow the same analysis pipeline to be executed for different assets.

# Conclusion

The initial data collection phase was successfully completed using historical BTC-USD data obtained from Yahoo Finance.

The dataset spans from 2018-01-01 to 2026-06-01 and contains 3,074 daily observations with no missing values detected. The available variables include Open, High, Low, Close, and Volume, providing sufficient information for future feature engineering and market regime analysis.

A MultiIndex column structure was identified during the inspection process. This structure is generated automatically by yfinance to support multi-asset downloads. Since the current analysis focuses on a single asset (BTC-USD), the second index level is redundant for this phase but remains an important consideration for future project scalability.

Based on the data quality assessment, the dataset is suitable for exploratory analysis and subsequent machine learning stages.